tool/grind/utils.dart

```diff
@@ -45,29 +44,33 @@ String environment(String name) {
 /// Ensure that the repository at [url] is cloned into the build directory and
 /// pointing to [ref].
 ///
+/// If [name] is passed, it's used as the basename of the directory for the
+/// repo. Otherwise, [url]'s basename is used.
+///
 /// Returns the path to the repository.
-Future<String> cloneOrCheckout(String url, String ref) async {
-  var name = p.url.basename(url);
-  if (p.url.extension(name) == ".git") name = p.url.withoutExtension(name);
+String cloneOrCheckout(String url, String ref, {String? name}) {
```

Comment:
Is there a reason why this is no longer asynchronous?

Reply:
We need to call it from within `pkg.npmAdditionalFiles.fn` which doesn't support async callbacks.